### DISCUSSION TITLE
Dotnet monitor memory usage collection rule

### DISCUSSION THREAD

**Discussion by tbajwa2017** [dotnet/dotnet-monitor#4271] [Q&A] [answered]
Hello,

There is collection rule for high cpu usage as below:

Collect Trace - High CPU Usage (EventCounter Trigger)

- name: DotnetMonitor_CollectionRules__HighCpuUsage__Trigger__Type
  value: "EventCounter"
- name: DotnetMonitor_CollectionRules__HighCpuUsage__Trigger__Settings__ProviderName
  value: "System.Runtime"
- name: DotnetMonitor_CollectionRules__HighCpuUsage__Trigger__Settings__CounterName
  value: "cpu-usage"
- name: DotnetMonitor_CollectionRules__HighCpuUsage__Trigger__Settings__GreaterThan
  value: "60"
- name: DotnetMonitor_CollectionRules__HighCpuUsage__Trigger__Settings__SlidingWindowDuration
  value: "00:00:10"
- name: DotnetMonitor_CollectionRules__HighCpuUsage__Actions__0__Type
  value: "CollectTrace"
- name: DotnetMonitor_CollectionRules__HighCpuUsage__Actions__0__Settings__Profile
  value: "Cpu"
- name: DotnetMonitor_CollectionRules__HighCpuUsage__Actions__0__Settings__Egress
  value: "artifacts"
- name: DotnetMonitor_CollectionRules__HighCpuUsage__Filters__0__Key
  value: "ProcessName"
- name: DotnetMonitor_CollectionRules__HighCpuUsage__Filters__0__Value
  value: "MyProcessName"

My question is, is there a collection rule for High Memory usage?

**Comment by jander-msft** [comment, marked as answer]
If I recall correctly, there is no counter for the entire memory usage of the application but there is one for the GC heap size (the total heap memory allocated for managed objects). We have an example of using this counter here: https://github.com/dotnet/dotnet-monitor/blob/main/documentation/collectionrules/collectionruleexamples.md#collect-gcdump---heap-size-eventcounter-trigger

**Reply by tbajwa2017** [reply]
Thanks Justin. That means we can use that heap memory allocated for dotnet apps as GC heap size since it is the memory allocated for all dotnet apps running and once the heap size is greater than value below, it will trigger the gcdump collection. In the case of container, there is only one app running with process ID 1.

- name: DotnetMonitor_CollectionRules__LargeGCHeapSize__Trigger__Settings__GreaterThan
  value: "10"
